storage/storage_interface.ml

```diff
@@ -118,9 +118,11 @@ type dp_stat_t = {
 
 let string_of_dp_stat_t (x: dp_stat_t) = Jsonrpc.to_string (rpc_of_dp_stat_t x)
 
+type device_config = (string * string) list
 type probe = {
-  srs: (string * sr_info) list; (* SRs we found *)
-  uris: string list; (* other uris we found which could be probed recursively *)
+  attachable: (device_config * sr_info) list; (* Attachable SRs we found *)
```

Comment:
Again, a question about the naming here: since in SMAPIv3 SM returns an `sr` for these SRs, which is the same type that's returned by `SR.attach`, does that mean that these SRs are already attached?
In SMAPIv3, the interface allows us to use these `sr`s returned by probe to do operations like `SR.ls`. :thinking:

Reply:
That is a good point, @MarkSymsCtx: does probe need to list already attached SRs differently from SRs that are present and not attached already?
Should we have 4 categories of probe results then?
* incomplete
* creatable
* attached
* introduceable/attachable

Reply:
Otherwise someone could take an SR returned by probe, run SR.ls on it as @gaborigloi  pointed out and get an error because its not actually mounted anymore...

Reply:
Yeah, maybe we could return 4 lists of probe results.

Reply:
I think we should definitely not return an `sr` for an existing SR that isn't attached. We should indicate differently that there is an existing SR with this configuration.